src/screens/EPub/controllers/file-builders/HTMLFileBuilder.js

```diff
@@ -85,56 +87,84 @@ class HTMLFileBuilder {
-      let curText = chapters[i].text;
-      let chapterTitleStart = curText.indexOf("<h2");
-      let chapterTitleEnd = curText.indexOf("</h2>");
-      let chapterTitle = curText.substring(chapterTitleStart, chapterTitleEnd);
-      chapterTitleStart = curText.indexOf(">");
-      chapterTitle = chapterTitle.substring(chapterTitleStart+1);
-      pdf.text(chapterTitle, 0, 10, 'left');
+    chapters.forEach((chapter, i) => {
+      let curText = chapter.text;
+      pdf.text(`${i+1} ${chapter.title}`, margin, 10, 'left');
       let pageCurrent = pdf.internal.getCurrentPageInfo().pageNumber;
-      pdf.outline.add(null, chapterTitle, {pageNumber:pageCurrent});
+      pdf.outline.add(null, chapter.title, {pageNumber:pageCurrent});
       let imgStart = curText.indexOf("src=");
       let imgEnd = curText.indexOf("alt=");
       let imgData = curText.substring(imgStart+5, imgEnd-2);
       pdf.addImage(imgData, 'JPEG', 15, 20, 180, 100);
-      pdf.text("Transcript", 0, 130, 'left');
+      // pdf.text("Transcript", 0, 130, 'left');
       let transcriptStart = curText.indexOf("<p>");
       let transcriptEnd = curText.indexOf("</p>");
+      let y = 140;
       if (transcriptStart !== -1) {
         let transcript = curText.substring(transcriptStart+3, transcriptEnd);
         let splitted = pdf.splitTextToSize(transcript, parseInt(w,10));
-        let y = 140;
-        for (let j = 0; j < splitted.length; j+=1) {
-          if (y >= h-h%10) {
-            y = 10;
-            pdf.addPage();
-          }
-          pdf.text(splitted[j], 0, y);
-          y += 10;
-        }
+        splitted.forEach(splitval => {
+            if (y >= h-h%10) {
+                y = 10;
+                pdf.addPage();
+            }
+            pdf.text(splitval, margin, y);
+            y += 10;
+        });
       }
+      chapter.subChapters.forEach((subchapter, j) => {
+          // Print subchapter title
+          y += 10;
+          pdf.text(`${i+1}-${j+1} ${subchapter.title}`, margin, y, 'left');
+
+          // Add subchapter to outline
+          pdf.outline.add(null, subchapter.title, {pageNumber:pageCurrent});
+
+          // Parse subchapter contents to produce transcript of text
+          subchapter.contents.forEach(subchText => {
+              if (typeof subchText === 'string') {
+                  // Add subchapter transcript
+                  let transcript = subchText.replace('\n', '').replace('\r', '').replace('#### Transcript', '')
+                  let splitted = pdf.splitTextToSize(transcript, parseInt(w,10));
+                  splitted.forEach(splitval => {
+                      if (y >= h-h%10) {
+                          y = 10;
+                          pdf.addPage();
+                      }
+                      pdf.text(splitval, margin, y);
+                      y += 10;
+                  });
+              } else {
+                  // TODO: should we insert/display images from subchapters?
```

Comment:
I'm betting that images should be included in subchapters. (we can test this)

Reply:
Absolutely! I'm happy to revisit this if we find that the images should be present there :+1: